pkg/config/mock.go

```diff
@@ -0,0 +1,27 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+package config
+
+import "strings"
+
+// MockConfig should only be used in tests
+type MockConfig struct {
+	Config
+}
+
+// Set is used for setting configuration in tests
+func (c *MockConfig) Set(key string, value interface{}) {
```

Comment:
Not sure if this really provides value since we already have `MergeConfig` exposed. I am thinking that most tests (at least for trace) would look like:
```go
config.Mock().MergeConfig(strings.NewReader(`
    api_key: test
    # ... more yaml
`)
```